editor/editor_inspector.cpp

```diff
@@ -3866,12 +3866,20 @@ void EditorInspector::set_use_filter(bool p_use) {
 void EditorInspector::register_text_enter(Node *p_line_edit) {
 	search_box = Object::cast_to<LineEdit>(p_line_edit);
 	if (search_box) {
-		search_box->connect(SceneStringName(text_changed), callable_mp(this, &EditorInspector::_filter_changed));
+		search_debouncer = memnew(Timer);
+		search_debouncer->set_wait_time(0.2);
+		search_debouncer->set_one_shot(true);
+		add_child(search_debouncer);
+
+		search_debouncer->connect("timeout", callable_mp(this, &EditorInspector::update_tree));
+		search_box->connect(SceneStringName(text_changed), callable_mp(this, &EditorInspector::_filter_changed).unbind(1));
 	}
 }
 
-void EditorInspector::_filter_changed(const String &p_text) {
-	update_tree();
+void EditorInspector::_filter_changed() {
+	if (search_debouncer->is_stopped()) {
+		search_debouncer->start();
```

Comment:
I wasn't sure if I should always restart timer. It's how it works in VS Code, but the search does not start until you stop writing. With the current condition, the inspector may update while you are typing, but it's still better than before.

Reply:
That gives me an idea 🤔